rllib/agents/ppo/ddppo.py

```diff
@@ -186,6 +202,122 @@ def validate_config(self, config):
         if config["kl_coeff"] != 0.0 or config["kl_target"] != 0.0:
             raise ValueError("DDPPO doesn't support KL penalties like PPO-1")
 
+    @override(PPOTrainer)
+    def setup(self, config: PartialTrainerConfigDict):
+        super().setup(config)
+
+        # Initialize torch process group for
+        if self.config["_disable_execution_plan_api"] is True:
+            ip = ray.get(self.workers.remote_workers()[0].get_node_ip.remote())
+            port = ray.get(self.workers.remote_workers()[0].find_free_port.remote())
+            address = "tcp://{ip}:{port}".format(ip=ip, port=port)
+            logger.info("Creating torch process group with leader {}".format(address))
+
+            # Get setup tasks in order to throw errors on failure.
+            ray.get(
+                [
+                    worker.setup_torch_data_parallel.remote(
+                        url=address,
+                        world_rank=i,
+                        world_size=len(self.workers.remote_workers()),
+                        backend=self.config["torch_distributed_backend"],
+                    )
+                    for i, worker in enumerate(self.workers.remote_workers())
+                ]
+            )
+            logger.info("Torch process group init completed")
+
+    @override(PPOTrainer)
+    def training_iteration(self) -> ResultDict:
+        # Shortcut.
+        first_worker = self.workers.remote_workers()[0]
+
+        # Run sampling and update steps on each worker in asynchronous fashion.
+        sample_and_update_results = asynchronous_parallel_requests(
+            remote_requests_in_flight=self.remote_requests_in_flight,
+            actors=self.workers.remote_workers(),
+            ray_wait_timeout_s=1000.0,  # 0.0
```

Comment:
huh?

Reply:
Ah, sorry, was just trying some stuff. Basically, this makes it synchronous :) Will revert. ...

Reply:
done

Reply:
Huh? Ok, now it's fixed ... Forgot to push.